admin/src/components/PresetsManager.js

```diff
@@ -1,101 +1,70 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import PropTypes from "prop-types";
 import { Table, TableBody, TableCell, TableHead, TableRow } from "@mui/material";
 import AddIcon from "@mui/icons-material/Add";
 import IconButton from "@mui/material/IconButton";
-import { PresetWidgetService } from "../services/presetWidget";
 import { PresetsTableRow } from "./PresetsTableRow";
+import { AbstractWidgetService } from "../services/abstractWidget";
 
-export class PresetsManager extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = { dataElements: [] };
-        this.service = this.props.service ?? new PresetWidgetService(this.props.apiPath, this.props.createErrorHandler);
-        this.loadData = () => {
-            this.service.loadPresets().then((result) => {
-                this.setState(state => ({
-                    ...state,
-                    dataElements: result,
-                }));
-            });
-        };
-        this.onCreate = (rowData = this.getDefaultRowData()) => {
-            return this.service.createPreset(rowData);
-        };
-        this.onEdit = (data) => {
-            console.log(data);
-            return this.service.editPreset(data.id, data.settings);
-        };
-        this.onDelete = (id) => {
-            return this.service.deletePreset(id);
-        };
-        this.onShow = ({ shown, id }) => {
-            return (shown ? this.service.hidePreset(id) : this.service.showPreset(id));
-        };
-    }
-
-    componentDidMount() {
-        this.loadData();
-        this.service.setReloadDataHandler(this.loadData);
-    }
-
-    getDefaultRowData() {
-        return this.props.tableKeys.reduce((ac, key) => ({ ...ac, [key]: "" }), {});
-    }
+export function DefaultAddPresetButton({ onCreate }) {
+    return (<IconButton color="primary" size="large" onClick={onCreate}><AddIcon/></IconButton>);
```

Comment:
codestyle: If you are passing props to another component stick to the names of the nested component. (onCreate => onClick) 
Or perhaps even pass all props for DefaultAddPresetButton to IconButton like so:
```javascript
export function DefaultAddPresetButton(props) {
return <IconButton color="primary" [...], {...props}><AddIcon/></IconButton>
}
```